templates/search_clients_daily_v7.sql

```diff
@@ -0,0 +1,153 @@
+WITH
+  augmented AS (
+  SELECT
+    *,
+    ARRAY_CONCAT( ARRAY(
+      SELECT
+        AS STRUCT element.source AS source,
+        element.engine AS engine,
+        element.count AS count,
+        CASE
+          WHEN (element.source IN ('searchbar',  'urlbar',  'abouthome',  'newtab',  'contextmenu',  'system',  'activitystream',  'webextension',  'alias') OR element.source IS NULL) THEN 'sap'
+          WHEN (STARTS_WITH(element.source, 'in-content:sap:')
+          OR STARTS_WITH(element.source, 'sap:')) THEN 'tagged-sap'
+          WHEN (STARTS_WITH(element.source, 'in-content:sap-follow-on:') OR STARTS_WITH(element.source,'follow-on:')) THEN 'tagged-follow-on'
+          WHEN STARTS_WITH(element.source, 'in-content:organic:') THEN 'organic'
+          WHEN STARTS_WITH(element.source, 'ad-click:') THEN 'ad-click'
+          WHEN STARTS_WITH(element.source, 'search-with-ads:') THEN 'search-with-ads'
+        ELSE
+        'unknown'
+      END
+        AS type
+      FROM
+        UNNEST(search_counts.list) ), ARRAY(
+      SELECT
+        AS STRUCT "ad-click:" AS source,
+        key AS engine,
+        value AS count,
+        "ad-click" AS type
+      FROM
+        UNNEST(scalar_parent_browser_search_ad_clicks.key_value) ), ARRAY(
+      SELECT
+        AS STRUCT "search-with-ads:" AS source,
+        key AS engine,
+        value AS count,
+        "search-with-ads" AS type
+      FROM
+        UNNEST(scalar_parent_browser_search_with_ads.key_value) ) ) AS _searches
+  FROM
+    main_summary_v4 ),
+  flattened AS (
+  SELECT
+    *
+  FROM
+    augmented,
+    UNNEST(
+    IF
+      (ARRAY_LENGTH(_searches) = 0,
```

Comment:
nice catch! took me a moment to figure out why this was needed. could you:
1. add a comment explaining the bug this avoids
2. use `_search` as the truthy result because `ARRAY_LENGTH(NULL) = 0` is falsey
   - not that `_searches` can be `NULL`, but just in case this gets modified in the future such that it can
3. replace the subquery expression with an array literal and an anonymous struct because:
   - a literal will have better performance
   - an anonymous struct will grab the field names from `_searches`, reducing repetition
4. replace `STRING(NULL)` with `CAST(NULL AS STRING)` because it's currently referencing the [`STRING(timestamp_expression)` function](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#string)
    - I'm filing a separate PR to explain this in `README.md`
```suggestion
      -- Provide replace empty _searches with one null search to ensure all clients are included in results
      (ARRAY_LENGTH(_searches) > 0,
        _searches,
        [(CAST(NULL AS STRING),
          CAST(NULL AS STRING),
          NULL,
          CAST(NULL AS STRING))])) ),
```